protocols/identify/src/handler.rs

```diff
@@ -213,8 +214,38 @@ impl Handler {
         }
     }
 
-    fn update_supported_protocols_for_remote(&mut self, remote_info: &Info) {
-        let new_remote_protocols = HashSet::from_iter(remote_info.protocols.clone());
+    fn handle_incoming_info(&mut self, info: Info) {
+        self.remote_info.replace(info.clone());
+
+        self.update_supported_protocols_for_remote(&info);
+
+        self.events
+            .push(ConnectionHandlerEvent::NotifyBehaviour(Event::Identified(
+                info,
+            )));
+    }
+
+    fn handle_incoming_push_info(&mut self, push_info: PushInfo) {
+        if let Some(mut info) = self.remote_info.take() {
+            info.merge(push_info);
+            self.remote_info.replace(info.clone());
+
+            self.update_supported_protocols_for_remote(&info);
+
+            self.events
+                .push(ConnectionHandlerEvent::NotifyBehaviour(Event::Identified(
+                    info,
+                )));
+        } else {
+            warn!(
```

Comment:
```suggestion
            log::debug!(
```

I don't think this warrants a `warn`.